### DISCUSSION TITLE
Bad request when using swagger and curl

### DISCUSSION THREAD

**Discussion by BrambleXu** [fastapi/fastapi#7841] [Questions] [answered]
I see this [article](https://towardsdatascience.com/how-to-deploy-a-machine-learning-model-dc51200fe8cf) and try the [code](https://github.com/MaartenGr/ML-API). But get bad request when using swagger and curl. The problem description is here https://github.com/MaartenGr/ML-API/issues/1

How should I saw the correct response with swagger and curl? Thanks.

**Comment by steinitzu** [comment, marked as answer]
In your linked issue you are sending invalid JSON to the API (keys and strings must be in double quotes). Fastapi handles this correctly by returning a "bad request" response.
The reason it works in the python script is because the JSON encoder formats your input as valid JSON, but curl and swagger UI just send exactly what you enter.

Replace all your single quotes with double quotes and it should work, e.g.

```json
{
    "satisfaction_level": 0.38,
    "salary": "low"
}
```

Instead of
```
{
    'satisfaction_level': 0.38,
    'salary': 'low'
}
```

**Comment by BrambleXu** [comment]
Thanks! The double-quote work!

**Comment by tiangolo** [comment]
Thanks for the help here @steinitzu ! :clap: :bow: 

Thanks for reporting back and closing the issue @BrambleXu  :+1:
